### PR TITLE
LigerKernel applied to LLM components for FT/DPO scripts

### PR DESCRIPTION
**Pull request by tyler-romero**
LigerKernel optionally applied to the LLM during Finetuning/DPO. Using these Kernels can improve throughput / memory efficiency.

See: https://github.com/allenai/open-instruct/issues/557

Many line-changes come directly from running `make style` on otherwise untouched files - happy to revert those if desired.
 
 Future Work:
 - Optionally use Liger chunked-loss for DPO, etc (these can provide massive reductions in memory requirements).
 - Optionally use Liger Fused linear cross-entropy for "sum" reduction in FT script

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ build-backend = "setuptools.build_meta"
 [tool.setuptools]
 py-modules = ["open_instruct"]
 
-# pytroch related setups
```
Comment (tyler-romero) [inline pyproject.toml:34]: fwiw, `uv` kept giving me errors with this pyproject.toml

**Comment by tyler-romero** [comment]
Tested by running the mini.sh scripts modified as:
```bash
accelerate launch \
    --mixed_precision bf16 \
    --num_processes 1 \
    open_instruct/finetune.py \
    --model_name_or_path Qwen/Qwen2.5-0.5B \
    --tokenizer_name Qwen/Qwen2.5-0.5B \
    --use_slow_tokenizer False \
    --use_flash_attn False \
    --use_liger_kernel True \
    --max_seq_length 1024 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --learning_rate 5e-06 \
    --lr_scheduler_type linear \
    --warmup_ratio 0.03 \
    --weight_decay 0.0 \
    --num_train_epochs 2 \
    --output_dir output/sft_qwen2p5-500M/ \
    --with_tracking \
    --report_to wandb \
    --logging_steps 1 \
    --reduce_loss sum \
    --model_revision main \
    --dataset_mixer_list allenai/tulu-3-sft-personas-algebra 100 \
    --add_bos \
    --seed 123
```

**Comment by tyler-romero** [comment]
Many line-changes come directly from running `make style` - happy to revert those if desired.

**Comment by tyler-romero** [comment]
Wanted to get some feedback on this before working on PPO or trying to incorporate liger's chunked losses

**Comment by hamishivi** [comment]
@vwxyzjn Feel free to review or not, this PR isn't really a massive change (just looks like it due to linting stuff) so I think it's fine to merge.

**Review by vwxyzjn — APPROVED** [review]
This LGTM! Feel free to merge. Thanks so much @tyler-romero!

**Comment by tyler-romero** [comment]
That checks out with my intuition @hamishivi, most of the kernels are fairly minor optimizations. The fused linear layer + loss kernels give big reductions in memory requirements but the easy out-of-the-box one is only wired up for pre-training or SFT.
